AU2/plugins/custom_plugins/ScoringPlugin.py

```diff
@@ -0,0 +1,252 @@
+Parameters:
+    a: attempts
+    b: bonus points
+    c: conkers
+    k: kills
+
+Syntax:
+    Currently the scoring formula is passed directly to the Python eval() function.
+    This means the syntax is that of a Python expression.
+    For reference, the arithmetic symbols are
+        + addition
+        - subtraction
+        * multiplication
+        / division
+        ** exponentiation
+    If you want to be fancy, `import math` is run right before the formula is evaluated,
+    so you can use functions from this module too.
+"""),
+            DefaultNamedSmallTextbox(identifier=self.html_ids["Formula"],
+                                     title="Scoring formula",
+                                     default=self.gsdb_get("Formula"))
+        ]
+
+    def answer_set_formula(self, html_response):
+        self.gsdb_set("Formula", html_response[self.html_ids["Formula"]])
+        return [Label("[SCORING] Set scoring formula.")]
+
+    def formula_is_valid(self, formula: Optional[str] = None) -> bool:
+        """
+        Checks whether the scoring formula is valid.
+
+        Parameters:
+            formula: the scoring formula in string form. Defaults to the value from the plugin.
+
+        Returns:
+            True if formula is valid, False otherwise
+        """
+        if formula is None:
+            formula = self.gsdb_get("Formula")
+        # since we don't have the formula parser yet,
+        # we crudely use a try-except evaluating the expression
+        a = 0
+        b = 0
+        c = 0
+        k = 0
+        import math
+        try:
+            eval(formula)
+            return True
+        except Exception:
+            return False
+
+    def on_page_generate(self, _) -> List[HTMLComponent]:
+        # don't generate open season page if open season hasn't started!
+        open_season_start = timestamp_to_dt(self.gsdb_get("Start"))
+        if not open_season_start or open_season_start >= get_now_dt():
+            return []
+
+        # also don't generate if formula is invalid
+        formula = self.gsdb_get("Formula")
+        if not self.formula_is_valid(formula):
+            return [Label("[WARNING] [SCORING] Invalid scoring formula -- skipping openseason page!")]
+
+        events = list(EVENTS_DATABASE.events.values())
+        score_manager = ScoreManager(ASSASSINS_DATABASE.get_identifiers(include=lambda a: not a.is_police),
+                                     formula=formula,
+                                     bonuses={
+                                         ident: self.aps_get(ident, "Bonus")
+                                         for ident in ASSASSINS_DATABASE.get_identifiers()
+                                     })
+        for e in events:
+            score_manager.add_event(e)
+
+        table_str = "Something went wrong..."
+        if score_manager.live_assassins:
+            # score_manager caches score, so calling twice is fine!

```

Comment:
You mentioned in one of the commits that caching is no longer the case (although it ends up not being important)